File: 476.java
public class Solution {
    public int findComplement(int num) {
        int count = 1;
        int cmp = 0;
        int temp = num;
        while(temp>0){
            temp = temp>>1;
            count++;
            cmp = (cmp<<1) + 1;
        }
        return num^cmp;
    }
}
// È¥Î» Âß¼­ÔËËã

File: 476.cpp
class Solution {
public:
    int findComplement(int num) {
		int input = num;
        int base = 1;
		int result = 0;
		while(input!=0){
			if(input%2==0)
				result = result + base;
			input/=2;
			base *= 2;
		}
		return result;
    }
};